connectgateway/get_namespace_test.py

```diff
@@ -0,0 +1,138 @@
+SERVICE_ACCOUNT_KEY = os.environ["GOOGLE_APPLICATION_CREDENTIALS"]
+ZONE = "us-central1-a"
+REGION = "us-central1"
+CLUSTER_NAME = f"cluster-{uuid.uuid4().hex[:10]}"
+
+
+@pytest.fixture(autouse=True)
+def setup_and_tear_down() -> None:
+    # nohing to setup here
+
+    # run the tests here
+    yield
+
+    try:
+        # delete the cluster
+        client = gke.ClusterManagerClient()
+        cluster_location = client.common_location_path(PROJECT_ID, ZONE)
+        cluster_name = f"{cluster_location}/clusters/{CLUSTER_NAME}"
+        op = client.delete_cluster({"name": cluster_name})
+        op_id = f"{cluster_location}/operations/{op.name}"
+
+    finally:
+        # schedule a retry to ensure the cluster is deleted
+        @backoff.on_predicate(
+            backoff.expo, lambda x: x != gke.Operation.Status.DONE, max_tries=20
+        )
+        def wait_for_delete() -> gke.Operation.Status:
+            return client.get_operation({"name": op_id}).status
+
+        wait_for_delete()
+
+
+def on_success(details: dict[str, str]) -> None:
+    """
+    A handler function to pass into the retry backoff algorithm as the function
+    to be executed upon a successful attempt.
+
+    Read the `Event handlers` section of the backoff python module at:
+    https://pypi.org/project/backoff/
+    """
+    print("Successfully created cluster after {elapsed:0.1f} seconds".format(**details))
+
+
+def on_failure(details: dict[str, str]) -> None:
+    """
+    A handler function to pass into the retry backoff algorithm as the function
+    to be executed upon a failed attempt.
+
+    Read the `Event handlers` section of the backoff python module at:
+    https://pypi.org/project/backoff/
+    """
+    print("Backing off {wait:0.1f} seconds after {tries} tries".format(**details))
+
+
+@backoff.on_predicate(
+    # the backoff algorithm to use. we use exponential backoff here
+    backoff.expo,
+    # the test function on the return value to determine if a retry is necessary
+    lambda x: x != gke.Operation.Status.DONE,
+    # maximum number of times to retry before giving up
+    max_tries=20,
+    # function to execute upon a failure and when a retry a scheduled
+    on_backoff=on_failure,
+    # function to execute upon a successful attempt and no more retries needed
+    on_success=on_success,
+)
+def poll_for_op_status(
+    client: gke.ClusterManagerClient, op_id: str
+) -> gke.Operation.Status:
+    """
+    This function calls the Operation API in GCP with the given operation id. It
+    serves as a simple retry function that fetches the operation and returns
+    it's status.
+
+    We use the 'backoff' python module to provide us the implementation of the
+    backoff & retry strategy. The function is annotated with the `backoff`
+    python module to schedule this function based on a reasonable backoff
+    algorithm.
+    """
+
+    op = client.get_operation({"name": op_id})
+    return op.status
```

Comment:
nit: I understand this was copied from another sample directly, but there is a lot of content here that makes this seem more complicated than it is. 

unrelated-nit: `backoff` package hasn't been updated in a number of years, doesn't state explicit support for versions after Python 3.10, and uses older str(format) style that isn't f-strings and can look "incorrect". 

Keeping the separate functions for the on_backoff and on_success as to not have to introduce lambdas
```suggestion
def on_success(details: dict[str, str]) -> None:
    print("Successfully created cluster after {elapsed:0.1f} seconds".format(**details))


def on_failure(details: dict[str, str]) -> None:
    print("Backing off {wait:0.1f} seconds after {tries} tries".format(**details))


@backoff.on_predicate(
    backoff.expo,
    lambda x: x != gke.Operation.Status.DONE,
    max_tries=20,
    on_backoff=on_failure,
    on_success=on_success,
)
def poll_for_op_status(
    client: gke.ClusterManagerClient, op_id: str
) -> gke.Operation.Status:
    op = client.get_operation({"name": op_id})
    return op.status
```

Reply:
Changed the polling function, please take a look.